Tests/Realm.Tests/Database/AsymmetricObjectTests.cs

```diff
@@ -0,0 +1,382 @@
+                {
+                    _ = asymmetribObj.PartitionLike;
+                }, "Attempted to access a detached row");
+            });
+        }
+
+        [Test]
+        public void AddSameAsymmetricObjTwice_DoesntThrow()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var flxConfig = await GetFLXIntegrationConfigAsync();
+                flxConfig.Schema = new[] { typeof(BasicAsymmetricObject) };
+                using var realm = await GetFLXIntegrationRealmAsync(flxConfig: flxConfig);
+                var partitionLike = Guid.NewGuid().ToString();
+                var asymmetricObj = new BasicAsymmetricObject
+                {
+                    PartitionLike = partitionLike
+                };
+
+                Assert.DoesNotThrow(() =>
+                {
+                    realm.Write(() =>
+                    {
+                        realm.Add(asymmetricObj);
+                    });
+
+                    realm.Write(() =>
+                    {
+                        realm.Add(asymmetricObj);
+                    });
+                });
+
+                await realm.SyncSession.WaitForUploadAsync();
+
+                var foundObjects = await GetObjFromRemoteThroughMongoClient<BasicAsymmetricObject>(
+                    flxConfig, nameof(BasicAsymmetricObject.PartitionLike), partitionLike);
+
+                Assert.That(foundObjects.Single().PartitionLike, Is.EqualTo(partitionLike));
+            });
+        }
+
+        [TestCaseSource(nameof(SetAndGetValueCases))]
+        [TestCaseSource(nameof(SetAndReplaceWithNullCases))]
+        public void SetAndRemotelyReadValue(string propertyName, object propertyValue)
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                Guid objId = Guid.Empty;
+                var flxConfig = await GetFLXIntegrationConfigAsync();
+                flxConfig.Schema = new[] { typeof(AsymmetricObjectWithAllTypes) };
+
+                using var realm = await GetFLXIntegrationRealmAsync(flxConfig: flxConfig);
+
+                realm.Write(() =>
+                {
+                    var asymmetricObjAllTypes = new AsymmetricObjectWithAllTypes { RequiredStringProperty = string.Empty };
+                    objId = asymmetricObjAllTypes.Id;
+                    TestHelpers.SetPropertyValue(asymmetricObjAllTypes, propertyName, propertyValue);
+                    realm.Add(asymmetricObjAllTypes);
+                });
+
+                await realm.SyncSession.WaitForUploadAsync();
+                var documents = await GetObjFromRemoteThroughMongoClient<AsymmetricObjectWithAllTypes>(
+                    flxConfig, propertyName, BsonValue.Create(propertyValue));
+
+                foreach (var doc in documents)
+                {
+                    Assert.That(TestHelpers.GetPropertyValue(doc, propertyName), Is.EqualTo(propertyValue));
+                }
```

Comment:
This assertion is weird. Since you're querying for objects with that property value, it's normal that all the returned documents will have the correct property. Instead, you should probably query by the objId and then assert on the property value.